core/vm/evm.go

```diff
@@ -469,14 +518,25 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	// Ensure there's no existing contract already at the designated address
 	contractHash := evm.StateDB.GetCodeHash(address)
 	if evm.StateDB.GetNonce(address) != 0 || (contractHash != (common.Hash{}) && contractHash != emptyCodeHash) {
+		// TODO this is a case where the witness should touch the already-created account
+		// however, it seems impossible to ever actually encounter this.
 		return nil, common.Address{}, 0, ErrContractAddressCollision
 	}
 	// Create a new account on the state
 	snapshot := evm.StateDB.Snapshot()
+
 	evm.StateDB.CreateAccount(address)
 	if evm.chainRules.IsEIP158 {
 		evm.StateDB.SetNonce(address, 1)
 	}
+	if evm.Accesses != nil {
+		// note: assumption is that the nonce, code size, code hash
+		// will be 0x0000...00 at the target account before it is created
+		// otherwise would imply contract creation collision which is
+		// impossible if self-destruct is removed
```

Comment:
unlikely != impossible

Reply:
actually the whole extension-and-suffix tree should not exist.

Reply:
A create/create2 cannot collide with an existing contract if we consider hash collisions to be impossible.  Is my understanding wrong?

Reply:
that's more or less right, except hash collision are possible, if only very unlikely. I'm nitpicking here.